modules/search/instant-search/lib/query-string.js

```diff
@@ -51,43 +51,47 @@ export function setSearchQuery( searchValue ) {
 	pushQueryString( encode( query ) );
 }
 
-const DEFAULT_SORT_MAP = {
-	'date|DESC': 'date_desc',
-	'date|ASC': 'date_asc',
-	'relevance|DESC': 'score_default',
-};
-
-// Convert a sort option like date|DESC to a sort key like date_desc
+const DEFAULT_SORT_MAP = new Map( [
+	// Old sort keys for backward compatibility
+	[ 'date|DESC', 'date_desc' ],
+	[ 'date|ASC', 'date_asc' ],
+	[ 'relevance|DESC', 'score_default' ],
+	// New sort keys from the customizer
+	[ 'newest', 'date_desc' ],
+	[ 'oldest', 'date_asc' ],
+	[ 'relevance', 'score_default' ],
+] );
+
+const INVERTED_SORT_MAP = new Map( [
```

Comment:
Is it worth doing this programmatically rather than maintaining two separate lists?

Reply:
I considered that, but programmatically inverting a map requires a bit of nuance since the keys must be unique. I figured it'd be simpler to maintain a separate list given that it's only 3 entries.